rest_framework_idempotency_key/migrations/0003_remove_user.py

```diff
@@ -3,21 +3,37 @@
 from django.db import migrations, models
 
 
+def remove_duplicated_records(apps, schema_editor):
+    IdempotencyKey = apps.get_model('rest_framework_idempotency_key', 'IdempotencyKey')
+    existed_keys = set()
+    for id_key in IdempotencyKey.objects.all():
+        if id_key.idempotency_key in existed_keys:
+            id_key.delete()
```

Comment:
2 questions:
- I'm wondering if it's safe to delete instances within a loop. Gemini told me it's unsafe but it doesn't provide a reference.
- I don't see any sorting options, how do you decide which one to delete when duplicated?

Reply:
1. 他擔心會發生 partial failure 吧，假設後面發生錯誤但是前面已經 delete 掉一些東西了。 
2. 對我們來說沒有差別 因為這個 duplication 源自於不同的 user，當我們砍掉 user 以後兩個 record 就視為同一個了，在這種狀況下不管砍掉哪一個都沒有差別。

Reply:
> 他擔心會發生 partial failure 吧
No, in most programming languages (even in Python), you shouldn't remove any item from a list during the iteration. It's because the removal changes the list itself.

> 實際上會重複 IdempotencyKey 的狀況我們發現也都幾乎是同時發出來的，是基於 bug 導致一個 user=None 的狀況使得他認成兩個
what if other columns are different? why not remove both?

Reply:
1. 這個說法蠻奇怪的，因為我不是從 list 裡面移除東西，這個描述是 AI 給的？？
2. 如果都砍掉的話我們 migrate 完以後有些還沒 expire 的 lock 瞬間會被我們清掉。 建議還是至少保留一個比較安全。